### PR TITLE
set next-auth and added @auth/core as optionalPeerDependencies, conve…

### LINKED ISSUE #6772: Missing peer next-auth@^4: Using Xata adapter in Sveltekit:
## Environment

```
System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M2 Pro
    Memory: 487.50 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
Binaries:
    Node: 19.6.0 - ~/.nvm/versions/node/v19.6.0/bin/node
    npm: 9.4.0 - ~/.nvm/versions/node/v19.6.0/bin/npm
Browsers:
    Brave Browser: 110.1.48.158
    Safari: 16.3
npmPackages:
    @auth/core: ^0.4.0 => 0.4.0
    @auth/sveltekit: ^0.2.1 => 0.2.1
    @next-auth/xata-adapter: ^0.2.2 => 0.2.2
    @sveltejs/kit: ^1.8.3 => 1.8.3
    @xata.io/client: ^0.22.1 => 0.22.1
```

### Reproduction URL

https://github.com/JustMrMendez/Xkelta-kit

### Unnecessary dependency *```next-auth```*

```
 WARN  Issues with peer dependencies found
.
└─┬ @next-auth/xata-adapter 0.2.2
  └── ✕ missing peer next-auth@^4
Peer dependencies that should be installed:
  next-auth@^4
```

### 1. Create a new Svelte project using the latest version and install @xata/cli:
   - `pnpm create svelte@latest my-app`
   - `pnpm install -g @xata.io/cli@latest`
   - `cd my-app`

### 2. A [xata.io](https://xata.io) account is required, login to CLI and create schema,
   - create a db schema using this JSON file [schema.json](https://github.com/JustMrMendez/issues/blob/main/schema.json)
   - `xata auth login`
   - `xata init --schema=schema.json`

### 3. Install dependencies:
   - `pnpm i @auth/core @auth/sveltekit @next-auth/xata-adapter @xata.io/client`

### 4. Add the required env variables for authjs
```bash
AUTH_SECRET=$(openssl rand -base64 32) && echo "AUTH_SECRET=$AUTH_SECRET" >> .env && echo "GITHUB_ID=<your github id>" >> .env && echo "GITHUB_SECRET=<your github secret>" >> .env
```

### 6. Add the following code to the src/hooks.server.ts file:
```TypeScript
import { SvelteKitAuth } from '@auth/sveltekit';
import GitHub from '@auth/core/providers/github';
import { GITHUB_ID, GITHUB_SECRET, XATA_API_KEY } from '$env/static/private';
import { XataAdapter } from '@next-auth/xata-adapter';
import { XataClient } from '$lib/xata';

const client = new XataClient({
    apiKey: XATA_API_KEY
});

export const handle = SvelteKitAuth({
    adapter: XataAdapter(client),
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore - SvelteKitAuth is not a function
    providers: [GitHub({ clientId: GITHUB_ID, clientSecret: GITHUB_SECRET })]
});
```
### 7. Add create +layout.server.ts file:
```typescript
import type { LayoutServerLoad } from './$types';

export const load = (async (event) => {
    return {
        session: await event.locals.getSession()
    };

}) satisfies LayoutServerLoad;
```

### 8. And finally, just add some auth logic to the home +page.svelte:
```Svelte
<script lang="ts">
	import { signIn, signOut } from '@auth/sveltekit/client';
	import { page } from '$app/stores';
</script>

<div>
  <h1>Let's get cracking bones!</h1>
    <p>
      {#if $page.data.session}
	{#if $page.data.session.user?.image}
	  <span style="background-image: url('{$page.data.session.user.image}')" class="avatar" />
	{/if}
	<span>
	  <small>Signed in as</small><br />
	  <strong>{$page.data.session.user?.name ?? 'User'}</strong>
	</span>
	<button on:click={() => signOut()} class="button">Sign out</button>
      {:else}
	<span>You are not signed in</span>
	<button on:click={() => signIn('github')}>Sign In with GitHub</button>
	{/if}
    </p>
</div>
```

Note: Replace the values of GITHUB_ID, GITHUB_SECRET, and XATA_API_KEY with your own values.
- [Create OAuth App](https://github.com/settings/developers) on GitHub to get them
 
This should reproduce the issue.

### Expected behavior

- It should not ask for `@next-auth` on install or warn of missing peer dependency
.
└─┬ @next-auth/xata-adapter 0.2.2
  └── ✕ missing peer next-auth@^4
- Or it should not work at all

## docs reference
- https://authjs.dev/reference/sveltekit
- https://authjs.dev/reference/adapters/xata

### PR DESCRIPTION
**Pull request by JustMrMendez**
…rted type Adapter import to "dynamic"

## ☕️ Reasoning
**Next-auth** is an optional dependency for projects using [**Authjs**](https://authjs.dev/).
the Xata adapter works with @auth-core too, so added both as `optionalPeerDependencies`

check https://github.com/nextauthjs/next-auth/issues/6772#issuecomment-1439337337 by @balazsorban44 and https://github.com/nextauthjs/next-auth/issues/6772#issuecomment-1440311502

## 🧢 Checklist

- [X] Documentation
- [ ] Tests
- [X] Ready to be merged

## 🎫 Affected issues

Fixes: #6772

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Let me rephrase my stance at https://github.com/nextauthjs/next-auth/issues/6772#issuecomment-1439337337.

I think `next-auth` should be added as a dependency for now, so we don't need complicated, hacky logic to resolve this issue.

When `@auth/core` is more stable, we will likely re-publish all adapters under the `@auth/*` namespace anyway, and they will use `@auth/core`'s `Adapters` interface instead.

**Comment by JustMrMendez** [comment]
> Let me rephrase my stance at [#6772 (comment)](https://github.com/nextauthjs/next-auth/issues/6772#issuecomment-1439337337).
> 
> I think `next-auth` should be added as a dependency for now, so we don't need complicated, hacky logic to resolve this issue.
> 
> When `@auth/core` is more stable, we will likely re-publish all adapters under the `@auth/*` namespace anyway, and they will use `@auth/core`'s `Adapters` interface instead.
